src/plugins/requestAuthPlugin.js

```diff
@@ -26,20 +21,34 @@ function verifyAuth([request, auth]) {
 		throw new Error(errorMessage);
 	}
 	// there are secret tokens in `auth`, be careful with logging
-	request.log(['auth'], `Authorizing with keys: ${JSON.stringify(keys)}`);
+	request.log(['info', 'auth'], `Authorizing with keys: ${JSON.stringify(keys)}`);
 }
 
-function injectAuthIntoRequest([request, auth]) {
-	// update request with auth info
-	request.state.oauth_token = auth.access_token;  // this endpoint provides 'access_token' instead of 'oauth_token'
-	request.state.refresh_token = auth.refresh_token;  // use to get new oauth upon expiration
-	request.state.expires_in = auth.expires_in;  // TTL for oauth token (in seconds)
-
-	// special prop in `request.app` to indicate that this is a new,
-	// server-provided token, not from the original request, so the cookies
-	// will need to be set in the response
-	request.app.setCookies = true;
-}
+/**
+ * Both the incoming request and the outgoing response need to have an
+ * 'authorized' state in order for the app to render correctly with data from
+ * the API, so this function modifies the request and the reply
+ */
+const applyAuth = ([request, auth]) => {
+	const path = '/';
+	const authState = {
+		oauth_token: {
+			value: auth.oauth_token || auth.access_token,
+			opts: { path, ttl: auth.expires_in * 1000 },
+		},
+		refresh_token: {
+			value: auth.refresh_token,
+			opts: { path, ttl: YEAR_IN_MS * 2 },
+		}
+	};
+	Object.keys(authState).forEach(name => {
+		const cookieVal = authState[name];
+		// apply to request
+		request.state[name] = cookieVal.value;
+		// apply to response - note this special `reply` prop assigned onPreAuth
+		request.authorize.reply.state(name, cookieVal.value, cookieVal.opts);
```

Comment:
This is bring up an interesting issue. In the current setup, we are assuming that authentication _never_ fails, because if the incoming request doesn't have any auth tokens, then new (anonymous) tokens are provided - that's the main function of the plugin currently.  

You are correct that the route handler will then attempt to render the app, but since there will be no API auth information in the request, that will fail with a `No auth info provided` Red Screen of Death.

It might actually be helpful to have a hard "Server Error" kind of response if/when auth fails _before_ attempting to render the app, just to contain the process a little more - I'll check out where that would go.

A related idea that I had while working on this is that we probably want to build some sort of 'retry' logic into this flow in case there's a temporary network problem when requesting auth tokens... it's pretty straightforward to implement that with RxJS, but it might be good to have some more discussion about the whole auth flow.